src/player.cpp

```diff
@@ -11030,6 +11065,8 @@ void player::remove_gunmod( item *weapon, unsigned id )
     }
     i_add_or_drop( *gunmod );
     weapon->contents.erase( weapon->contents.begin() + id );
+
+    moves -= gunmod->type->gunmod->install_time / 2;
```

Comment:
The line above removes the gunmod item from the weapons content. Any dereferencing of `gunmod` after that leads to undefined behavior. The moves should be deducted before the item is removed.